src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+                        if self._shutdown_flag.is_set():
+                            logging.debug("Shutting down listening thread...")
+                            return
+                        msg += self._socket.recv(4096)
+                        time.sleep(0.01)
+
+                    decoded_msg = Message.from_bytes(msg)
+                    logging.debug("R: Msgtype: %s", decoded_msg)
+                    self._actOnMessage(decoded_msg)
+                
+            except socket.timeout:
+                # just keep waiting for more messages.
+                pass
+
+            except Exception as e:
+                logging.error(e)
+
+            finally:
+                time.sleep(0.01)
+
+        # shutdown flag set....
+        logging.debug("Shutting down listening thread...")
+
+    def _actOnMessage(self,decoded_msg):
+        '''
+        Act on a message received. 
+        decoded_msg: A message dictionary that was decoded with self._decode
+        '''
+        if isinstance(decoded_msg, Message_EXPLAINED_STATUSMSG):
+            # Received a status message with an explanation string.
+            # Acknowledge it and log it.
+            self._sendMessage(Message_ENCODED_STATUSREPLY(decoded_msg.ecStatus))
+            logging.debug(decoded_msg.strExplanation)
+            err_status = decoded_msg.ecStatus
+
+            if err_status < 0:
+                self.state._set_value(HwError(decoded_msg.strExplanation), force_write=True)
+                self._measuring_flag.clear()
+                raise HwError("Error %d: %s" % (err_status, ERRCODE_DICT[err_status]))
+
+        elif isinstance(decoded_msg, Message_ENCODED_STATUSMSG):
+            # Received a status message with an encoded status
+            # Acknowledge it and log it
+
+            # Check status messages
+            err_status = decoded_msg.ecStatus
+            self._sendMessage(Message_ENCODED_STATUSREPLY(decoded_msg.ecStatus))
+
+            if err_status == PQ_ERRCODE_NO_ERROR:
+                # the measurement completed successfully.
+                self.state._set_value(model.ST_RUNNING, force_write=True)
+                    
+            elif err_status == PQ_ERRCODE_MEASUREMENT_READY:
+                # Measurement is complete
+                if self._measuring_flag.is_set():
+                    logging.info("Acquisition completed successfully.")
+                    self._measuring_flag.clear()
+                    # update metadata and send data array to subscribers
+                    md = self._metadata.copy()
+                    md[model.MD_DWELL_TIME] = self.scanner.dwellTime.value
+                    md[model.MD_ACQ_DATE] = self._tStart
+                    da = model.DataArray([[0]], md)
+                    self.data.notify(da)
+
+            else:
+                # All other status types denote errors. 
+                logging.error("Error %d: %s", err_status, ERRCODE_DICT[err_status])
```

Comment:
Why not raising HwError here, while in the other places where you receive an error, you do that.